libs/payments/paypal/src/lib/paypalCustomer/paypalCustomer.repository.ts

```diff
@@ -0,0 +1,154 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import {
+  AccountDatabase,
+  PaypalCustomerUpdate,
+  NewPaypalCustomer,
+} from '@fxa/shared/db/mysql/account';
+
+import {
+  PaypalCustomerManagerError,
+  PaypalCustomerNotUpdatedError,
+} from './paypalCustomer.error';
+
+/**
+ * Creates a paypalCustomer in the database.
+ *
+ * @returns The created paypalCustomer or throws an error if the paypalCustomer if it couldn't be created
+ */
+export async function createPaypalCustomer(
+  db: AccountDatabase,
+  paypalCustomer: NewPaypalCustomer
+) {
+  await db
+    .insertInto('paypalCustomers')
+    .values(paypalCustomer)
+    .executeTakeFirstOrThrow();
+  return fetchPaypalCustomer(
+    db,
+    paypalCustomer.uid,
+    paypalCustomer.billingAgreementId
+  );
+}
+
+/**
+ * Fetch paypalCustomer from the database by PK (uid & billingAgrementId).
+ *
+ * @returns Fetched paypalCustomer or throws an error if that paypalCustomer does not exist
+ */
+export async function fetchPaypalCustomer(
+  db: AccountDatabase,
+  uid: Buffer,
+  billingAgreementId: string
+) {
+  return db
+    .selectFrom('paypalCustomers')
+    .where('uid', '=', uid)
+    .where('billingAgreementId', '=', billingAgreementId)
+    .selectAll()
+    .executeTakeFirstOrThrow();
+}
+
+/**
+ * Fetch paypalCustomers from the database by uid.
+ *
+ * @returns Fetched paypalCustomers
+ */
+export async function fetchPaypalCustomersByUid(
+  db: AccountDatabase,
+  uid: Buffer
+) {
+  return db
+    .selectFrom('paypalCustomers')
+    .where('uid', '=', uid)
+    .orderBy('createdAt', 'desc')
+    .selectAll()
+    .execute();
+}
+
+/**
+ * Fetch paypalCustomers from the database by billingAgreementId.
+ *
+ * @returns Fetched paypalCustomers
+ */
+export async function fetchPaypalCustomersByBillingAgreementId(
+  db: AccountDatabase,
+  billingAgreementId: string
+) {
+  return db
+    .selectFrom('paypalCustomers')
+    .where('billingAgreementId', '=', billingAgreementId)
+    .selectAll()
+    .execute();
+}
+
+/**
+ * Update a paypalCustomer in the database with a given updated values.
+ *
+ * Note that the paypalCustomer passed in is not representative of the database after
+ * this function is called. If the updated paypalCustomer is needed, it should be fetched
+ * from the database again.
+ *
+ * @returns true if the paypalCustomer was updated, false if the update failed
```

Comment:
Does this return `false`? Looks like it might just be `true` or `throw`.

Reply:
It can return false, yep. L109.